### ISSUE TITLE
No Graphical LUKS2 password prompt at boot

### ISSUE THREAD

**Issue by TriMoon** [ublue-os/bazzite#831]
### Describe the bug

This is a crosspost of: https://discord.com/channels/1072614816579063828/1212374800908030032

---
Since a few updates it seems the graphical password prompt _(from plymouth)_ you get to unlock the LUKS2 partition, is broken.

Just a few moments ago, i updated see https://discord.com/channels/1072614816579063828/1212371178832793721, and after first reboot the graphical prompt got back.
__BUT__ after another reboot, it was back to a text only prompt again with a gray screen and 3 dot animation after entering the password.

```console
mainuser@uBlue:~$ rpm-ostree status 
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-nvidia:latest
                   Digest: sha256:b2e9d91aee99cb448274955238093f129e7871aa6506c25e20ca6b735ac31c2f
                  Version: 39.20240116.0 (2024-02-25T07:26:47Z)
          LayeredPackages: obfs4 tor
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf"' 

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-nvidia:latest
                   Digest: sha256:e7bf4f3cc4ae67e49f81cf1d6a4f57a90656757cde71c02be4e267eff689b418
                  Version: 39.20240116.0 (2024-02-24T02:02:28Z)
          LayeredPackages: obfs4 tor
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf"'
```

---
This still is hapening after each update, see my latest `rpm-ostree status` below.

### What did you expect to happen?

That the password prompt would stay in graphical mode.

### Output of `rpm-ostree status`

```shell
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-nvidia:latest
                   Digest: sha256:dc61eefec28e6173732dcac3411eb4e0a993e0eb2c8ee31fbe92c8b92666c6cb
                  Version: 39.20240116.0 (2024-02-29T06:17:33Z)
          LayeredPackages: obfs4 tor
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf"' 

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-nvidia:latest
                   Digest: sha256:b2e9d91aee99cb448274955238093f129e7871aa6506c25e20ca6b735ac31c2f
                  Version: 39.20240116.0 (2024-02-25T07:26:47Z)
          LayeredPackages: obfs4 tor
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf"'
```

### Hardware

https://forum.linux-hardware.org/index.php?topic=123.0

**Comment by dylanmtaylor** [comment]
Do you have a non-Nvidia GPU in the system?

**Comment by p5** [comment]
I faced the same on my Silverblue image.  Started about a month ago, but I haven't tried to fix it.

I feel something may have changed in upstream Fedora

**Comment by dylanmtaylor** [comment]
On my system with an Intel iGPU and an Nvidia GPU, it works fine, on Nvidia only it does not and it falls back to text and the three dots.

**Comment by TriMoon** [comment]
> @dylanmtaylor 
> Do you have a non-Nvidia GPU in the system?

Yes the `Ryzen9 7950X3D` CPU has an iGPU inside...

**Comment by TriMoon** [comment]
# UPDATE:

Today i tried something, which might shed a light upon this problem...
- I plugged my monitor into the HDMI connector on the mainboard, which is **the output of the iGPU**, and rebooted the graphical password prompt was there.
- When i plugged my monitor back into the HDMI connector on the nVidia card, and rebooted the password prompt was back to text only.

So the graphical password prompt will only be shown at boot when not using the nVidia card's output, which indicates the driver is not properly loaded at that time.
_(I'm booting with SecureBoot enabled and have the akmods MOK key enrolled...)_

**Comment by KyleGospo** [comment]
Fixed as of 2.5.0
